### PR TITLE
Filters now apply to detail view. 

### LINKED ISSUE #38: Move definition of "No AE" flag to config
The tool requires placeholder rows for participants with no AEs. Those participants are identified by a custom keyword in the user-specified `major` and `minor` column. Currently the keywords are hard-coded in `/src/chart/util/prepareData.js`. We should move them to `defaultSettings.js` instead and let the user overwrite the default as desired.

### PR DESCRIPTION
**Pull request by jwildfire**
Core issue (#5) led to a few additional refactor items. Closes #38 and #43

### CODE REVIEW

**Comment by jwildfire** [comment]
@samussiah The last round of updates is in. Had to do a bit of work on the data flow to get the new filters to work well. See [this commit](https://github.com/RhoInc/aeexplorer/pull/45/commits/833bb5a8b5ef8149330b89e05665d7d9fc035675) for details.

File: src/chart/defaultSettings.js [hunk 1]
```diff
@@ -23,7 +23,9 @@ const defaultSettings =
     ,'groups':
         []
     ,'defaults':
-        {'maxPrevalence': 0
+        {
+        'placeholderFlag':{'value_col':"AEBODSYS", 'values':["NA"]}
```
Comment (samussiah) [inline src/chart/defaultSettings.js]: placeholderFlag is showing up in the detail listing.

File: src/chart/AETable/init.js [hunk 2]
```diff
@@ -108,6 +106,13 @@ export function init(chart) {
   /////////////////////////////////////
   // Draw the summary table headers.
   /////////////////////////////////////
+    //Check to make sure there is some data
+    if (!dataMajor.length) {
+        chart.wrap.select('.SummaryTable').append('div').attr('class', 'alert')
+        .text("Error: No data matches the current filters. Update the filters to see results.");
```
Comment (samussiah) [inline src/chart/AETable/init.js]: I almost had a heart attack!

File: src/chart/detailTable/init.js [hunk 3]
```diff
@@ -60,7 +60,7 @@ export function init(chart,detailTableSettings) {
             'Details for ' + details.length + ' <b>' + minor + ' (' + major + ')</b> records' );
 
   //Details about current population filters
-    var filtered = chart.raw_data.length != chart.population_data.length
+    var filtered = chart.raw_event_data.length != chart.population_event_data.length
```
Comment (samussiah) [inline src/chart/detailTable/init.js]: Nice.

File: src/chart/init.js [hunk 4]
```diff
@@ -9,11 +9,20 @@ export function init(data) {
...
-  //Initialize adverse event eplorer.
+    //settings and defaults
     this.util.setDefaults(this)
     this.layout();
+
+    //Flag placeholder rows in raw data save a separate event-only data set
+    var placeholderValues = this.config.defaults.placeholderFlag.values
+    var placeholderCol = this.config.defaults.placeholderFlag.value_col
+    this.raw_data.forEach(d => d.placeholderFlag = placeholderValues.indexOf(d[placeholderCol])> -1)
+    this.raw_event_data = data.filter(d => !d.placeholderFlag)
```
Comment (samussiah) [inline src/chart/init.js]: Love it!

File: src/chart/util/prepareData.js [hunk 5]
```diff
@@ -1,44 +1,40 @@
 /*------------------------------------------------------------------------------------------------\
   Filter the raw data per the current filter and group selections.
+  After this function is executed there should be 4 data objects bound to the chart:
```
Comment (samussiah) [inline src/chart/util/prepareData.js]: Awesome.

**Comment by jwildfire** [comment]
@samussiah Fixed the `placeholderFlag` issue. Merging.

**Pull request merged**